core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -2277,6 +2277,29 @@ class DAGSchedulerSuite extends SparkFunSuite with LocalSparkContext with Timeou
       (Success, 1)))
   }
 
+  test("task end event should have updated accumulators (SPARK-20342)") {
+    val accumIds = new HashSet[Long]()
+    val listener = new SparkListener() {
+      override def onTaskEnd(event: SparkListenerTaskEnd): Unit = {
+        event.taskInfo.accumulables.foreach { acc => accumIds += acc.id }
+      }
+    }
+    sc.addSparkListener(listener)
+
+    // Try a few times in a loop to make sure. This is not guaranteed to fail when the bug exists,
+    // but it should at least make the test flaky. If the bug is fixed, this should always pass.
+    (1 to 10).foreach { _ =>
+      accumIds.clear()
+
+      val accum = sc.longAccumulator
+      sc.parallelize(1 to 10, 10).foreach { _ =>
```

Comment:
The context in these tests uses the "local" master, which has a single core, to you'd have only one task running at a time. So it should be the same, no?